Part1_InterruptControllersEvolution.html

```diff
@@ -34,30 +34,30 @@ <h3>Introduction</h3>
 
 	<img src="https://habrastorage.org/webt/le/yf/6e/leyf6enogyzp6flpijo2d0x1w9g.png" />
 
-	<p>Soon 8 lines weren't enough. To increase total interrupt line number two controllers 8259 (master and slave) were connected in a cascade (Dual PIC).<p/>
+	<p>Soon 8 lines weren't enough. To increase the total number of interrupt lines two 8259 controllers (master and slave) were connected in a cascade (Dual PIC).<p/>
 
-	<p>IRQs from 0 to 7 are processed with the first Intel 8259 PIC (master), and IRQs from 8 to 15 are processed with the second Intel 8259 PIC (slave). Only master is connected to CPU and can signal about the incoming interrupt. If there is an interrupt on lines 8-15, the second PIC (slave) will signal about it to master on the line IRQ2, and after that master will make a signal to CPU. This cascaded interrupt takes 1 from 16 lines, but makes total of 15 interrupts for all external devices.<p/>
+	<p>IRQs from 0 to 7 are processed with the first Intel 8259 PIC (master), and IRQs from 8 to 15 are processed with the second Intel 8259 PIC (slave). Only the master is connected to the CPU and can signal about the incoming interrupts. If there is an interrupt on lines 8-15, the second PIC (slave) will signal about it to the master on the line IRQ2, and after that the master will signal the CPU. This cascaded interrupt takes away 1 of the 16 lines, but makes a total of 15 interrupts for all external devices.<p/>
 
 	<img src="https://habrastorage.org/webt/xj/yn/dh/xjyndhabujjwz0alrbx831zfzt8.png" />
 
-	<p>This scheme was approved by a community and when now someone talks about PIC (Programm Interrupt Controller) they mean this Dual PIC system. After some time controllers 8259 have gotten some improvements and got a new name 8259A. And this scheme was included in chipset. At a time when the main bus for external devices connection was ISA bus, this system was enough. It was only necessary to that differenet devices did not connect to the same IRQ line, as this produces conflicts cause ISA interrupts aren't shareable.<p/>
+	<p>This scheme was adopted by the community, and now when someone talks about PIC (Programm Interrupt Controller) they mean this Dual PIC system. After some time the 8259 controllers were improved and got a new name: 8259A. In these controllers the DUAL PIC system was included in the chipset. At a time when the main bus for external device connection was the ISA, this system was sufficient. It was only necessary that different devices did not connect to the same IRQ line, since ISA interrupts aren't shareable.<p/>

```

Comment:
"In these controllers the DUAL PIC system was included in the chipset" - I can't really understand this phrase. Maybe "With these 8259A controllers the Dual PIC system was included in the chipset" ?

Reply:
Yeah that's better

Reply:
```suggestion
	<p>This scheme was adopted by the community, and now when someone talks about PIC (Programm Interrupt Controller) they mean this Dual PIC system. After some time the 8259 controllers were improved and got a new name: 8259A. With these controllers the DUAL PIC system was included in the chipset. At a time when the main bus for external device connection was the ISA, this system was sufficient. It was only necessary that different devices did not connect to the same IRQ line, since ISA interrupts aren't shareable.<p/>
```